refactor(#134, #121): 버튼, 라벨, 홈 리뷰 컴포넌트 리팩토링

suubiin opened the pull request:
### refactor(#134, #121): 버튼, 라벨, 홈 리뷰 컴포넌트 리팩토링

### PR 유형

- [x] CSS 등 사용자 UI 디자인 변경
- [x] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩터링
- [x] 주석 추가 및 수정


### PR 체크 리스트

<!-- PR이 다음 요구 사항을 충족하는지 확인하세요. -->

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트)

### 🔎 작업 내용

<!-- 어떤 작업을 하셨는지 상세하게 작성해주세요. -->

- 버튼 컴포넌트: 버튼 타입 프롭으로 설정, handleClick, restProps 프롭 추가
- 찜 버튼 컴포넌트:  버튼 타입 프롭으로 설정, onToggle, restProps 프롭 추가
- 홈 리뷰 컴포넌트: 대체 택스트 프롭 삭제, Link 태그 안에 article 태그 추가, h1 태그->h3 태그, h2 태그->p 태그 수정
- 공통: 변수 타입 지정, 컴포넌트 사용법 주석 추가, 웹 접근성 고려 속성 추가
- 스토리북 코드 수정
  <br />

### 🔧 앞으로의 과제

- 내일 할 일을 적어주세요.
  <br />

### 이슈

<!-- 이슈 키워드와 함께 #을 입력한 후 이슈 번호를 선택해주세요. -->
<!-- 예시 : resolves #1 -->

resolves #134, #121

suubiin committed:
refactor: 버튼, 라벨, 홈 리뷰 컴포넌트 리팩토링

- 버튼 컴포넌트: 버튼 타입 프롭으로 설정, handleClick, restProps 프롭 추가
- 찜 버튼 컴포넌트:  버튼 타입 프롭으로 설정, onToggle, restProps 프롭 추가
- 홈 리뷰 컴포넌트: 대체 택스트 프롭 삭제, Link 태그 안에 article 태그 추가, <h1>-><h3>, <h2>-><p> 수정
- 공통: 변수 타입 지정, 컴포넌트 사용법 주석 추가, 웹 접근성 고려 속성 추가
- 스토리북 코드 수정

woolen-h commented on src/components/Buttons/Buttons.tsx:
```diff
@@ -1,114 +1,164 @@
+   content: string;
+   type?: 'button' | 'submit' | 'reset';
+   handleClick: () => void;
+   [props: string] : any;
```
```
export interface ButtonDefaultProps {
   content: string;
   type?: 'button' | 'submit' | 'reset';
   handleClick: () => void;
   isError?: boolean;
   isLarge?: boolean;
   fullWidth?: boolean;
   [props: string] : any;
}
```

혹시 요런 느낌으로 isError, isLarge를 속성으로 전달해서 관리할 수 있을까용...?!

woolen-h commented on src/components/Buttons/Buttons.tsx:
```diff
@@ -1,114 +1,164 @@
-   text?: string;
+// 사용법
+// <ButtonDefault content="버튼 내용" type="button" handleClick={() => console.log('Clicked')} />
+export interface ButtonDefaultProps {
```
```
export interface ButtonDefaultProps {
   content: string;
   type?: 'button' | 'submit' | 'reset';
   handleClick: () => void;
   isError?: boolean;
   isLarge?: boolean;
   fullWidth?: boolean;
   [props: string] : any;
}
```

혹시 요런 느낌으로 isError, isLarge, FullWidth를 속성으로 전달해서 관리할 수 있을까용...?!

happyhye reviewed (approved):
검토 완료! 👍

minnje reviewed (approved):
✅

suubiin committed:
refactor: 버튼 컴포넌트 리팩토링
- ButtonDefault, ButtonFullWidth, ButtonLarge 컴포넌트를 하나의 Button 컴포넌트에 props 지정 방식으로 수정
- 수정사항 반영하여 스토리북 수정
